src/ed25519.rs

```diff
@@ -593,6 +593,91 @@ impl ExpandedSecretKey {
 
         Signature{ r: r, s: s }
     }
+
+    /// Sign a `prehashed_message` with this `ExpandedSecretKey` using the
+    /// Ed25519ph algorithm defined in [RFC8032 §5.1][rfc8032].
+    ///
+    /// # Inputs
+    ///
+    /// * `prehashed_message` is an instantiated hash digest with 512-bits of
+    ///   output which has had the message to be signed previously fed into its
+    ///   state.
+    /// * `public_key` is a [`PublicKey`] which corresponds to this secret key.
+    /// * `context` is an optional context string, up to 255 bytes inclusive,
+    ///   which may be used to provide additional domain separation.  If not
+    ///   set, this will default to an empty string.
+    ///
+    /// # Returns
+    ///
+    /// An Ed25519ph [`Signature`] on the `prehashed_message`.
+    ///
+    /// [rfc8032]: https://tools.ietf.org/html/rfc8032#section-5.1
+    pub fn sign_prehashed<D>(&self,
+                             prehashed_message: D,
```

Comment:
Why does `sign_prehashed` take a digest instance, instead of just taking the prehash (as a `[u8; 64]` or whatever?)

The digest instance is immediately consumed to produce the prehash, so there's no benefit internal to the function from taking a digest, but it makes the API less flexible, since the computation of the hash value is still coupled to the signing.

Reply:
My vague reasoning was that I wanted to force people to actually digest things with an appropriately sized hash, versus all the other things they could do if it just took a `[u8; 64]`.

Reply:
*waves arms in a squiggly fashion* ~~ T Y P E   S A F E T Y~~